feature-libs/requested-delivery-date/root/components/delivery-mode-date-picker/delivery-mode-date-picker.component.ts

```diff
@@ -0,0 +1,152 @@
+ */
+
+import { HttpErrorResponse } from '@angular/common/http';
+import { Component, OnDestroy, OnInit, Optional } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { Cart } from '@spartacus/cart/base/root';
+import { CheckoutSupportedDeliveryModesQueryReloadEvent } from '@spartacus/checkout/base/root';
+import {
+  CxDatePipe,
+  ErrorModel,
+  EventService,
+  GlobalMessageService,
+  GlobalMessageType,
+  TranslationService,
+} from '@spartacus/core';
+import { Card, OutletContextData } from '@spartacus/storefront';
+import { Observable, Subscription } from 'rxjs';
+import { filter, map } from 'rxjs/operators';
+import { RequestedDeliveryDateFacade } from '../../facade/requested-delivery-date.facade';
+
+@Component({
+  selector: 'cx-request-delivery-date',
+  templateUrl: './delivery-mode-date-picker.component.html',
+  providers: [CxDatePipe],
+})
+export class DeliveryModeDatePickerComponent implements OnInit, OnDestroy {
+  constructor(
+    protected datePipe: CxDatePipe,
+    protected requestedDelDateFacade: RequestedDeliveryDateFacade,
+    protected eventService: EventService,
+    protected translation: TranslationService,
+    protected globalMessageService: GlobalMessageService,
+    @Optional() protected deliveryOutlet?: OutletContextData
+  ) {}
+
+  protected cartEntry: Cart = {};
+  protected subscription = new Subscription();
+
+  protected earliestRetrievalAt: string | undefined;
+  protected requestedRetrievalAt: string | undefined;
+  protected form: FormGroup = new FormGroup({
+    requestDeliveryDate: new FormControl(),
+  });
+  protected isDatePickerReadOnly: boolean = true;
+
+  ngOnInit(): void {
+    if (this.deliveryOutlet?.context$) {
+      this.subscription.add(
+        this.deliveryOutlet.context$.subscribe((context) => {
+          this.cartEntry = context?.item;
+          this.isDatePickerReadOnly = context?.readonly || false;
+        })
+      );
+    }
+
+    if (this.isEarliestRetrievalDatePresent()) {
+      this.earliestRetrievalAt = this.cartEntry.earliestRetrievalAt;
+    }
+    if (this.isRequestedDeliveryDatePresent()) {
+      this.requestedRetrievalAt = this.cartEntry.requestedRetrievalAt;
+    } else {
+      //set the value of requestedRetrievalAt as earliestRetrievalAt and update occ.
+      this.requestedRetrievalAt = this.earliestRetrievalAt;
+      this.setRequestedDeliveryDate();
+    }
+    this.form.patchValue({
+      requestDeliveryDate: this.requestedRetrievalAt,
+    });
+  }
+
+  isEarliestRetrievalDatePresent(): boolean {
+    return this.isDateStringPresent(this.cartEntry?.earliestRetrievalAt);
+  }
+
+  isRequestedDeliveryDatePresent(): boolean {
+    return this.isDateStringPresent(this.cartEntry?.requestedRetrievalAt);
+  }
+
+  isDateStringPresent(value: string | undefined): boolean {
+    return value != null && value !== undefined && value.length > 0;
+  }
+
+  getRequestedDeliveryDateCardContent(
```

Comment:
Can we move this to common place ? Order-overview as well uses the same ?

Reply:
good point.

Noted. Will create a helper class to reuse code.

Reply:
Hi @vigneshkrish8691, added the date validation logic to a separate service.

Though we use ` getRequestedDeliveryDateCardContent(` in both the places, the return  parameters are different. For `DeliveryModeDatePickerComponent`, both `[textTitle, isoDate]` are a part of `text`. But in `OrderOverviewDeliveryDateComponent`, they are separate as we need both `text` and `title`